 design/README.MD

```diff
@@ -0,0 +1,456 @@
+## Overview
+
+Smithy is an interface definition language.
+It lets you abstract your interface
+from any given executable implementation.
+By focusing on the abstract interface
+Smithy helps you build an interface
+that can safely evolve
+maintaining both forward and backwards compatibility.
+
+Dafny is a programming language with formal verification features.
+Dafny code is translated to higher-level languages
+to provide interoperability for your verified code
+like Java or C#.
+These native representations are then executable.
+This means that a single Dafny program
+can be translated into multiple interoperable programs.
+
+Dafny itself has a `string` type
+that you might assume can map
+to the obvious string type in each translated language.
+
+However a native `string` may be UTF8, UTF16/USC2, or UTF32 
+depending on the language.
+Numbers are similarly complicated.
+Some languages have different sized numbers,
+while others have either a different default or a single size.
+There are also complicated types like streams.
+All these languages have equivalent
+but not congruent types to each other.
+
+In this project Dafny and Smithy are combined
+to generate concrete interfaces in Dafny
+and supported Dafny translation targets.
+These generated interfaces all use the same translated Dafny implementation
+and expose native interfaces and types for each language.
+
+The Smithy shapes are the common denominator
+between all these different types/conversions.
+
+Concretely our traditional greeting program:
+`() => "Hello World!"`
+generate interfaces that can be used like this:
+1. Dafny -> `var hello: seq<char> :- greet();`
+1. Java -> `String hello = greet();`
+1. C# -> `string hello = greet();`
+1. Python -> `hello = greet()`
+1. Rust -> `let hello: String = greet();`
+
+## Definitions
+
+- Runtime or native runtime:
```

Comment:
i'm new to the project so might be lacking some context - but isn't "runtime" a super overloaded term? why not just call them "transpilation targets"?